kiosk/src/Components/EnterHighScore.tsx

```diff
@@ -11,31 +11,87 @@ interface IProps {
   }
 
 const EnterHighScore: React.FC<IProps> = ({ kiosk }) => {
-    const [initialIndexRef, ] = useState(new PrimitiveRef(0));
-    const [initialIndex, setInitialIndex] = useState(initialIndexRef.value);
+    const [indexChanged, setIndexChanged] = useState(false);
     const [initials, setInitials] = useState(Array(configData.HighScoreInitialsLength + 1).join(configData.HighScoreInitialAllowedCharacters[0]));
+    const [timesAPressed, setTimesAPressed] = useState(0);
+    const [runOnce, setRunOnce] = useState(false);
+    const [firstRun, setFirstRun] = useState(true);
+
+    const existingHighScores = kiosk.getHighScores(kiosk.selectedGame!.id);
+    const aboveScores = existingHighScores.filter(item => item.score > kiosk.mostRecentScores[0]);
+    const belowScores = existingHighScores.slice(aboveScores.length, existingHighScores.length);
 
     if (!kiosk.mostRecentScores || !kiosk.mostRecentScores.length) {
         throw new Error("Cannot load high score entry view without having recent scores");
     }
 
-    const renderList = (highScores: HighScore[], offset: number): JSX.Element[] => {
-        return highScores.map((highScore, i) => {
-            return (
-                <tr className="highScoreRow" key={i}>
-                    <td className="highScoreIndex">{i + offset}.</td>
-                    <td className="highScoreInitials">
-                        <span className="highScoreInitial">{highScore.initials[0]}</span>
-                        <span className="highScoreInitial">{highScore.initials[1]}</span>
-                        <span className="highScoreInitial">{highScore.initials[2]}</span>
-                    </td>
-                    <td className="highScoreInitialsScoreSpacer"></td>
-                    <td className="highScoreScore">{highScore.score}</td>
-                </tr>
-            )
-        })
+    const gamepadLoop = () => {
+        if (kiosk.state !== KioskState.EnterHighScore) { return; }
+
+        if (kiosk.gamepadManager.isAButtonPressed()) {
+            setIndexChanged(true);
+        } else {
+            setIndexChanged(false);
+        }
+
+        if (kiosk.gamepadManager.isBButtonPressed()) {
```

Comment:
Looks like this escapes out and skips high score input -- it feels like it might be nice to make it so we allow you to fix typos using b, no reason to be locked to the frustrating mechanics of arcade machine (I'm guessing that mechanic was partially that way in the first place just to get people to leave faster so someone else would put in another quarter).

Could keep the skip out mechanic if nothing has been entered?

Reply:
I like the thought of fixing with B, I'll see what I can do. 

For skipping if nothing's been entered, would this just look like timing out after a certain amount of time?

Reply:
I just the current behavior - if no input is locked in by pressing A, and b is pressed, you could choose to make it exit without saving as it looks like it is currently set to do. I just meant that as an option to support both behaviors, not something I particularly wanted to see. (Could also use menu button as the escape hatch)

A timer is also something common / useful to make sure it doesnt stay on high score forever if the person leaves, but I'd probably count it as separate work personally

Reply:
Oh, I see. Basically, make sure to save the high score even if the user didn't elect to change the initials.

Reply:
That could be a nice thing too but I didn't intend to add any more work here with that statement -- I just meant that if you wanted the back out option it currently has even when adding in 'b to fix typos', it could be changed to something like 

```typescript
if (player has pressed a and made letter selections) {
     // remove the last selection so they can fix their typo
} else {
    // player hasn't made any selections at all and is pressing back so escape
    kiosk.navigate(KioskState.GameOver);
}
```

and support both~

Reply:
Yeah, that makes sense. I just split the two things out, though, as it should be anyway. Hitting the B buttons shouldn't take you out of the screen, it should just impact initials that you're inputting. 

I now have the escape button being the sole thing that exits.